src/mon/MonmapMonitor.cc

```diff
@@ -611,6 +611,17 @@ bool MonmapMonitor::prepare_command(MonOpRequestRef op)
       err = -EINVAL;
       goto reply_no_propose;
     }
+    if (monmap.stretch_mode_enabled) {
+      for (const auto &p : loc) {
+        if (!mon.osdmon()->osdmap.crush->name_exists(p.second)) {
+          ss << "location doesn't belong to any existing crush buckets!"
+          << " If you are trying to replace an arbiter mon, please use the command:"
+          <<  " ceph mon set_new_tiebreaker";
```

Comment:
Don't you need to add the tiebreaker before you can configure it as the tiebreaker? Or did we move that to be a special command to avoid all this?

Reply:
You're right the output is misleading, you need to do `ceph mon add` and then  `ceph mon set_new_tiebreaker`. I'll just remove that part of the output.

Reply:
I mean the problem is the tiebreaker monitor's bucket cannot exist in the crush map (or at least, there can't be OSDs there, so having the map reflect it is not idiomatic). And the documentation would need to be changed even if accepting that solution.

Reply:
And once in stretch mode you are blocked from adding monitors without a location, if that wasn't clear. So this commit actually prevents replacing the tiebreaker at all.